src/config.cc

```diff
@@ -351,6 +351,9 @@ void Config::updateFromCommandLine(const boost::program_options::variables_map&
 
   if (cmd.count("legacy-interface") != 0) {
     boost::filesystem::path legacy_interface = cmd["legacy-interface"].as<boost::filesystem::path>();
+    if (!boost::filesystem::exists(legacy_interface)) {
+      throw std::runtime_error(std::string("legacy-interface ") + legacy_interface.string() + " is not exists!!!");
+    }
```

Comment:
This check already exists in checkLegacyVersion. It just logs an error instead of throwing an exception. I think we should just move the runtime error line to that function.